system/Autoloader/Autoloader.php

```diff
@@ -306,10 +306,16 @@ public function sanitizeFilename(string $filename): string
         // Plus the forward slash for directory separators since this might be a path.
         // http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_278
         // Modified to allow backslash and colons for on Windows machines.
-        $filename = preg_replace('/[^0-9\p{L}\s\/\-\_\.\:\\\\]/u', '', $filename);
+        $tmp = preg_replace('/[^0-9\p{L}\s\/\-\_\.\:\\\\]/u', '', $filename);
 
         // Clean up our filename edges.
-        return trim($filename, '.-_');
+        $cleanFilename = trim($tmp, '.-_');
+
+        if ($filename !== $cleanFilename) {
+            throw new InvalidArgumentException('The file path contains special character that is not allowed: "' . $filename . '"');
```

Comment:
I think for better developer experience, the exception message would contain which disallowed characters were included in the filename. Like `The file path contains special characters "()" that are not allowed.`. Maybe use `preg_match_all` and implode the `$matches` into a single string, something like that?

Reply:
Improved exception message.